valhalla/midgard/util.h

```diff
@@ -472,6 +471,29 @@ struct ring_queue_t {
   const_iterator end() const { return v.end(); }
 };
 
+struct gps_segment_t {
+  std::vector<midgard::PointLL> shape;
+  float speed; //in meters/second
+};
+
+/**
+ * Generate a gps path from an original path (with speeds) adding in some random noise
+ *
+ * @param segments    The representation of the segments of the original path including a speed per segment
+ *                    this is usually maneuvers or edges in a route path generated by thor
+ * @param accuracies  The returned minimum accuracy per simulated point, that is the simulated point should
+ *                    be no more than accuracy from the original point (after resampling, see below)
+ * @param smoothing   Controls the variability of adjacent simulated points by smoothing the error over a
+ *                    window of simulated points. In this case smoothing determines the window size
+ * @param accuracy    The maximum noise (arc distance) an one simulated point can exhibit with respect to
+ *                    the original point (after resampling, see below)
+ * @param sample_rate The final sample rate for the returned points. The determines how much time,
+ *                    as a function of the segments speed, passes between individual simulated points
+ */
+std::vector<midgard::PointLL> simulate_gps(
```

Comment:
move this back to the libvalhalla api as its pretty useful for other things. but leave out the boost dep in favor of a custom object